pkg/api/steve/collections/collections.go

```diff
@@ -0,0 +1,699 @@
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	resourceClient := h.dynamicClient.Resource(resource)
+	opts := metav1.ListOptions{}
+	paramCodec.DecodeParameters(r.URL.Query(), metav1.SchemeGroupVersion, &opts)
+	resources, err := resourceClient.List(r.Context(), opts)
+	if apierrors.IsNotFound(err) {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	returnResp(w, resources.UnstructuredContent())
+}
+
+// unscopedHandler handles requests for resources in namespaces that don't belong to any project.
+func (h *handler) unscopedHandler(w http.ResponseWriter, r *http.Request) {
+	logrus.Tracef("[%s] handling request %s\n", apiServiceName, r.URL.Path)
+	vars := mux.Vars(r)
+	op := vars["op"]
+	var projectsOrNamespaces []string
+	if _, ok := vars[projectsOrNamespacesVar]; ok {
+		projectsOrNamespaces = strings.Split(vars[projectsOrNamespacesVar], ",")
+	}
+	namespaces := make(map[string]*corev1.Namespace)
+	resource, err := h.gvrFromVars(vars)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	resourceClient := h.dynamicClient.Resource(resource)
+	if len(projectsOrNamespaces) > 0 {
+		for _, n := range projectsOrNamespaces {
+			ns, _ := h.namespaceCache.Get(n)
+			namespaces[n] = ns
+		}
+	}
+	// get namespaces that don't have the projectID label
+	selector, _ := labels.Parse("!" + nslabels.ProjectIDFieldLabel)
+	nss, err := h.namespaceCache.List(selector)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	finalNamespaces := make(map[string]*corev1.Namespace)
+	for _, ns := range nss {
+		// no namespaces in the selector so all can be included
+		if len(namespaces) == 0 {
+			finalNamespaces[ns.Name] = ns
+			continue
+		}
+		// only include resources in namespaces specified in the selector
+		if op != "!" {
+			if n, ok := namespaces[ns.Name]; ok && n != nil {
+				finalNamespaces[ns.Name] = ns
+			}
+		} else { // only include resources in namespaces NOT specified in the selector
+			if _, ok := namespaces[ns.Name]; !ok {
+				finalNamespaces[ns.Name] = ns
+			}
+		}
+	}
+	namespaces = finalNamespaces
+
+	resp, err := h.getResourcesForNamespaces(r, resource, resourceClient, namespaces)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	returnResp(w, resp)
+}
+
+// scopedHandler handles requests for resources in namespaces in one particular project.
+// steve sends requests to individual namespace endpoints based on the user's rolebindings,
+// so this endpoint has to handle both projects and namespaces in the selector even though
+// only namespaces in the one project will be returned.
+func (h *handler) scopedHandler(w http.ResponseWriter, r *http.Request) {
+	logrus.Tracef("[%s] handling request %s\n", apiServiceName, r.URL.Path)
+	vars := mux.Vars(r)
+	project := vars["project"]
```

Comment:
I'm a bit confused on how this value interacts with the `pjorns` value. It seems like we are offering two methods to filter - one that allows filtering for an arbitrary number of namespaces/projects, and one that offers filtering for one project. 

Given that the second filter includes the first, I'm not sure that having both of them makes sense in this context.

Reply:
See my response to Kevin and the updated doc comment, this complexity is a requirement for compatibility with steve.

Reply:
Resolving this, was explained offline.